File: cloud/oauth-server/src/main/java/pers/acp/admin/oauth/base/BaseDomain.java
package pers.acp.admin.oauth.base;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Sort;
import org.springframework.transaction.annotation.Transactional;
import pers.acp.admin.common.po.QueryParam;
import pers.acp.admin.oauth.entity.User;
import pers.acp.admin.oauth.repo.UserRepository;
import pers.acp.core.CommonTools;

/**
 * @author zhang by 26/12/2018
 * @since JDK 11
 */
@Transactional(readOnly = true)
public class BaseDomain {

    protected final UserRepository userRepository;

    @Autowired
    public BaseDomain(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public User findCurrUserInfo(String loginNo) {
        return userRepository.findByLoginno(loginNo).orElse(null);
    }

    public User findUserById(String id) {
        return userRepository.findById(id).orElse(null);
    }

    protected PageRequest buildPageRequest(QueryParam queryParam) {
        if (CommonTools.isNullStr(queryParam.getOrderName())) {
            return PageRequest.of(queryParam.getCurrPage() - 1, queryParam.getPageSize());
        } else {
            Sort.Direction direction = Sort.Direction.DESC;
            if (queryParam.getOrderCommond().equalsIgnoreCase("asc")) {
                direction = Sort.Direction.ASC;
            }
            return PageRequest.of(queryParam.getCurrPage() - 1, queryParam.getPageSize(), direction, CommonTools.toCamel(queryParam.getOrderName()).split(","));
        }
    }

}

File: common/acp-admin-cloud-dependencies/src/main/kotlin/pers/acp/admin/common/base/BaseDomain.kt
package pers.acp.admin.common.base

import io.github.zhangbinhub.acp.core.CommonTools
import org.springframework.data.domain.PageRequest
import org.springframework.data.domain.Sort
import pers.acp.admin.common.po.QueryParam

/**
 * @author zhang by 15/01/2019
 * @since JDK 11
 */
abstract class BaseDomain {

    protected fun buildPageRequest(queryParam: QueryParam): PageRequest =
        if (CommonTools.isNullStr(queryParam.orderName) || CommonTools.isNullStr(queryParam.orderCommand)) {
            PageRequest.of(queryParam.currPage!! - 1, queryParam.pageSize!!)
        } else {
            var direction: Sort.Direction = Sort.Direction.ASC
            if (queryParam.orderCommand!!.startsWith("desc", ignoreCase = true)) {
                direction = Sort.Direction.DESC
            }
            PageRequest.of(
                queryParam.currPage!! - 1,
                queryParam.pageSize!!,
                direction,
                *queryParam.orderName!!.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            )
        }

}
